services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -1034,6 +1039,22 @@ public URL toUrl(@NonNull String accessToken) {
     if (metadata() != null) {
       sb.append(String.format("&metadata=%s", metadata()));
     }
+    if (unrecognized() != null) {
+      for (Map.Entry<String, SerializableJsonElement> entry : unrecognized().entrySet()) {
+        JsonElement element = entry.getValue().getElement();
+        if (element.isJsonPrimitive()) {
+          sb.append(String.format("&%s=%s", entry.getKey(), element.getAsString()));
+        } else {
+          throw new IllegalStateException(
```

Comment:
```suggestion
          throw new IllegalArgumentException(
```

Reply:
`unrecognized()` isn't an argument, it's a state

Reply:
It was developer's argument and that's the origin of the problem. In any case, this isn't really important.